count-words/test.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+	"unicode"
+)
+
+func IsSep(r rune) bool { return unicode.IsSpace(r) || unicode.IsPunct(r) || unicode.IsSymbol(r) }
+
+func wordOccurrences(s string) (words []string, occurrences []uint) {
+	var w []string
+	var tmp []rune
+	wasSep := true
+	for _, r := range s {
+		if IsSep(r) {
+			if !wasSep {
+				w = append(w, string(tmp))
+				// tmp = tmp[:0]
+				tmp = nil
+			}
+			wasSep = true
+		} else {
+			wasSep = false
+			tmp = append(tmp, r)
+		}
+	}
+	if len(tmp) > 0 {
+		w = append(w, string(tmp))
+	}
```

Comment:
Did you copy it from `words` function that we wrote during the lesson? Why wouldn't you just call the function `words` then?